velox/exec/tests/TableWriteTest.cpp

```diff
@@ -228,40 +291,258 @@ TEST_F(TableWriteTest, constantVectors) {
+                 [&](auto row) {
+                   return StringView(fmt::format("str_{}", row));
+                 }),
+             makeFlatVector<int64_t>(
+                 numPartitions, [&](auto row) { return row + 1000; })});
+      });
   createDuckDbTable(vectors);
 
+  auto inputFilePaths = makeFilePaths(numBatches);
+  for (int i = 0; i < numBatches; i++) {
+    writeToFile(inputFilePaths[i]->path, vectors[i]);
+  }
+
   auto outputDirectory = TempDirectoryPath::create();
   auto plan = PlanBuilder()
-                  .tableScan(rowType_)
+                  .tableScan(rowType)
                   .tableWrite(
-                      rowType_->names(),
+                      rowType->names(),
                       std::make_shared<core::InsertTableHandle>(
                           kHiveConnectorId,
                           makeHiveInsertTableHandle(
-                              rowType_->names(),
-                              rowType_->children(),
-                              {},
+                              rowType->names(),
+                              rowType->children(),
+                              partitionKeys,
                               makeLocationHandle(outputDirectory->path))),
-                      CommitStrategy::kTaskCommit,
+                      CommitStrategy::kNoCommit,
                       "rows")
                   .project({"rows"})
                   .planNode();
 
-  assertQuery(plan, filePaths, "SELECT count(*) FROM tmp");
+  auto task = assertQuery(plan, inputFilePaths, "SELECT count(*) FROM tmp");
+
+  // Verify partition directories 1:1 correspond to partition names.
+  std::set<std::string> actualPartitionDirectories =
+      getLeafSubdirectories(outputDirectory->path);
 
-  // HiveTaskCommitWriteProtocol writes to dot-prefixed file in the
-  // outputDirectory which is still picked up by table scan.
+  std::set<std::string> expectedPartitionDirectories;
+  std::set<std::string> partitionNames;
+  for (auto i = 0; i < vectors.size(); i++) {
+    for (auto row = 0; row < vectors[i]->size(); row++) {
+      auto partitionName = makePartitionName(vectors[i], {1, 3}, row);
+      expectedPartitionDirectories.emplace(
+          fs::path(outputDirectory->path) / partitionName);
+      partitionNames.emplace(partitionName);
+    }
+  }
+  EXPECT_EQ(actualPartitionDirectories, expectedPartitionDirectories);
+
+  // Verify distribution of records in partition directories.
+  auto iterPartitionDirectory = actualPartitionDirectories.begin();
+  auto iterPartitionName = partitionNames.begin();
+  while (iterPartitionDirectory != actualPartitionDirectories.end()) {
+    assertQuery(
+        PlanBuilder().tableScan(rowType).planNode(),
+        makeHiveConnectorSplits(*iterPartitionDirectory),
+        fmt::format(
+            "SELECT * FROM tmp WHERE {}",
+            partitionNameToPredicate(*iterPartitionName)));
+    // Files written to each partition directory is per driver.
```

Comment:
What is 'task->numOutputDrivers()' here? Is it 1 or not? If it is 1. then we are not testing that there is a file per driver.

Reply:
Yes, task->numOutputDrivers() is now always 1 for Hive connector, after debugging into the code. So for tests I will check `EXPECT_EQ(countRecursiveFiles(*iterPartitionDirectory), 1)` for now.

(Originally I thought it's easier to switch from 1 to 1+ max drivers. It turns out it needs more code changes.)